lib/cylc/gui/app_gcylc.py

```diff
@@ -1305,8 +1307,13 @@ def get_right_click_menu(self, task_ids, t_states, task_is_family=False):
                     item.set_image(gtk.image_new_from_stock(
                         gtk.STOCK_DND, gtk.ICON_SIZE_MENU))
                     view_menu.append(item)
-                    item.connect('activate', self.view_task_info,
-                                 task_ids[0], filename)
+                    if is_graph_view:
+                        item.connect('button-release-event',
```

Comment:
Rather than repeating the logic we could do something like:
```python
if is_graph_view:
    connect = lambda item, x, y, z: item.connect('button-relase-event', x, y, z)
else:
    connect = lambda item, x, y, z: item.connect('activate', x, None, y, z)

connect(item, self.view_task_info, task_ids[0], filename)
```